liiweb/views/legislation.py

```diff
@@ -8,25 +8,43 @@
 
 from peachjam.helpers import chunks, get_language
 from peachjam.models import JurisdictionProfile, Legislation, Locality, pj_settings
-from peachjam_api.serializers import LegislationSerializer
+from peachjam.views import FilteredDocumentListView
 
 
-class LegislationListView(TemplateView):
+class LegislationListView(FilteredDocumentListView):
     template_name = "liiweb/legislation_list.html"
     variant = "current"
     navbar_link = "legislation"
     model = Legislation
+    queryset = Legislation.objects.prefetch_related(
+        "taxonomies", "taxonomies__topic", "work"
+    )
     extra_context = {"doc_table_citations": True, "legislation_list_sort": "title"}
 
     def get_queryset(self):
+        qs = super().get_queryset()
         qs = (
-            self.model.objects.exclude(published=False)
-            .distinct("work_frbr_uri")
-            .order_by("work_frbr_uri", "-date", "language__pk")
+            qs.exclude(published=False)
+            .distinct(
+                "title",
```

Comment:
The distinct is required to ensure we only get the latest expression. I'm not sure how we also get ordering by title into that for pagination.